added rybbit service template

zehjotkah opened the pull request:
## Changes
- added rybbit service template

## Issues
- https://github.com/coollabsio/coolify/pull/5835

zehjotkah committed:
added rybbit service template

ShadowArcanist commented on templates/compose/rybbit.yaml:
```diff
@@ -0,0 +1,138 @@
+      - 'NEXT_PUBLIC_DISABLE_SIGNUP=${DISABLE_SIGNUP:-false}'
+    depends_on:
+      - rybbit_backend
+    restart: unless-stopped
```
Coolify adds restart policy automatically so this is not needed
```suggestion
```

ShadowArcanist commented on templates/compose/rybbit.yaml:
```diff
@@ -0,0 +1,138 @@
+
+volumes:
+  postgres_data:
+  clickhouse_data:
```
Coolify adds this automatically so this is not needed
```suggestion
```

ShadowArcanist commented on templates/compose/rybbit.yaml:
```diff
@@ -0,0 +1,138 @@
+      timeout: 10s
+      retries: 3
+      start_period: 10s
+    restart: unless-stopped
```
Coolify adds restart policy automatically so this is not needed
```suggestion
```

ShadowArcanist commented on templates/compose/rybbit.yaml:
```diff
@@ -0,0 +1,138 @@
+      interval: 30s
+      timeout: 10s
+      retries: 3
+    restart: unless-stopped
```
Coolify adds restart policy automatically so this is not needed
```suggestion
```

ShadowArcanist commented on templates/compose/rybbit.yaml:
```diff
@@ -0,0 +1,138 @@
+
+  rybbit_clickhouse:
+    image: 'clickhouse/clickhouse-server:25.4.2'
+    container_name: rybbit-clickhouse
```
Coolify adds container names automatically so this is not needed
```suggestion
```

ShadowArcanist commented on templates/compose/rybbit.yaml:
```diff
@@ -0,0 +1,138 @@
+
+  rybbit_postgres:
+    image: 'postgres:17.4'
+    container_name: rybbit-postgres
```
Coolify adds container names automatically so this is not needed
```suggestion
```

ShadowArcanist commented on templates/compose/rybbit.yaml:
```diff
@@ -0,0 +1,138 @@
+
+  rybbit_backend:
+    image: 'ghcr.io/rybbit-io/rybbit-backend:latest'
+    container_name: rybbit-backend
```
Coolify adds container names automatically so this is not needed
```suggestion
```

ShadowArcanist commented on templates/compose/rybbit.yaml:
```diff
@@ -0,0 +1,138 @@
+services:
+  rybbit:
+    image: 'ghcr.io/rybbit-io/rybbit-client:latest'
+    container_name: rybbit-client
```
Coolify adds container names automatically so this is not needed
```suggestion
```

ShadowArcanist commented on templates/compose/rybbit.yaml:
```diff
@@ -0,0 +1,138 @@
+      timeout: 10s
+      retries: 3
+      start_period: 10s
+    restart: unless-stopped
```
Coolify adds restart policy automatically so this is not needed
```suggestion
```

ShadowArcanist commented on templates/compose/rybbit.yaml:
```diff
@@ -0,0 +1,138 @@
+      # --- Internal Config ---
+      - 'CLICKHOUSE_HOST=http://rybbit_clickhouse:8123'
+      - CLICKHOUSE_USER=default
+      - CLICKHOUSE_DB=analytics
```
Since the user and db is already hard coded on `rybbit_clickhouse` service we don't have to hard code it here (in future if we need to change these value we just need to change in one place)
```suggestion
      - CLICKHOUSE_USER=${CLICKHOUSE_USER}
      - CLICKHOUSE_DB=${CLICKHOUSE_DB}
```

ShadowArcanist commented on templates/compose/rybbit.yaml:
```diff
@@ -0,0 +1,138 @@
+      - POSTGRES_HOST=rybbit_postgres
+      - POSTGRES_PORT=5432
+      - POSTGRES_DB=analytics
+      - POSTGRES_USER=frog
```
Since the user and db is already hard coded on `rybbit_postgres` service we don't have to hard code it here (in future if we need to change these value we just need to change in one place)

```suggestion
      - POSTGRES_DB=${POSTGRES_DB}
      - POSTGRES_USER=${POSTGRES_USER}
```

ShadowArcanist commented on templates/compose/rybbit.yaml:
```diff
@@ -0,0 +1,138 @@
+services:
+  rybbit:
+    image: 'ghcr.io/rybbit-io/rybbit-client:latest'
+    container_name: rybbit-client
+    environment:
+      - NODE_ENV=production
+      - 'NEXT_PUBLIC_BACKEND_URL=${SERVICE_URL_RYBBIT}'
+      - 'NEXT_PUBLIC_DISABLE_SIGNUP=${DISABLE_SIGNUP:-false}'
```
The `port: 3000` mentioned on the comments at the top of the compose is useful for users who are using `caddy` as their Coolify Proxy but those who use `traefik` as their Coolify Proxy might see `no available server` error because of the missing `SERVICE_URL_RYBBIT_3000`

The `SERVICE_URL_RYBBIT_3000` also tells Coolify to automatically generate a sub domain for the service with port `3000` at the end so the user just have to change the domain to the domain they prefer to use but the auto generated domain will work out of the box (if they have DNS configured properly to resolve all of their sub domains to the server where they are deploying this service)

```suggestion
    environment:
      - SERVICE_URL_RYBBIT_3000
      - NODE_ENV=production
      - 'NEXT_PUBLIC_BACKEND_URL=${SERVICE_URL_RYBBIT}'
      - 'NEXT_PUBLIC_DISABLE_SIGNUP=${DISABLE_SIGNUP:-false}'
```

ShadowArcanist reviewed (commented):
Hey @zehjotkah, thanks for the PR!

This looks great, especially for your first PR on GitHub. Definitely better than most service template PRs I’ve seen! 

I’ve left a few suggestions to improve the service compose, so please check those out when you get a chance.

Also, it seems like this hasn't been tested yet (based on your comment [here](https://github.com/coollabsio/coolify/pull/5835#issuecomment-3299196509). If that’s the case, could you please test it thoroughly and let us know?

Additionally, it would be awesome if you could add a PR to the docs repo here: [coolify-docs](https://github.com/coollabsio/coolify-docs). You can take a look at [this PR](https://github.com/coollabsio/coolify-docs/pull/352/files) for an idea of which files to edit or add.

zehjotkah committed:
Update rybbit.yaml

- fixed port
- removed unnecessary stuff (container_name, restart)
- added client/frontend healthcheck
- DRY for hardcoded values

zehjotkah commented:
Thanks, I updated the yaml with your suggestions. Thanks for teaching me. 
Also I fixed the port and added a health check for the frontend. 
Do you know a way to add the generated frontend URL also to the backend but add /api?  
Yes, I tested it. Just thought that others might want to test it, too.

zehjotkah commented:
added pull request for the docs: 
[coollabsio/coolify-docs#363](https://github.com/coollabsio/coolify-docs/pull/363)

ShadowArcanist commented:
> Thanks, I updated the yaml with your suggestions. Thanks for teaching me. Also I fixed the port and added a health check for the frontend. Do you know a way to add the generated frontend URL also to the backend but add /api? Yes, I tested it. Just thought that others might want to test it, too.

Cool!

To set the backend URL with the `/api` path, you can use:

```yaml
- BACKEND_URL=$SERVICE_URL_FRONTEND/api
```

This appends `/api` to the auto-generated frontend URL and assigns it to the `BACKEND_URL` environment variable. It's a good approach because if the user changes the domain via the UI, the backend URL will automatically update to reflect that change.

I put together a quick Docker Compose file to test this setup:

```yaml
services:
  frontend:
    image: 'nginx:alpine'
    environment:
      - SERVICE_URL_FRONTEND_3000
    command: >
      /bin/sh -c "echo \"Frontend URL: $SERVICE_URL_FRONTEND_3000\" && nginx -g 'daemon off;'"

  backend:
    image: 'nginx:alpine'
    environment:
      - BACKEND_URL=$SERVICE_URL_FRONTEND/api
    command: >
      /bin/sh -c "echo \"Backend URL: $BACKEND_URL\" && nginx -g 'daemon off;'"
```

And here’s the log output showing the value of the environment variable:
<img width="1546" height="1244" alt="image" src="https://github.com/user-attachments/assets/3c4f14cc-7334-45c9-bdc0-2f8c45369061" />


I only configured the domain for the frontend service in the UI:
<img width="1556" height="1109" alt="image" src="https://github.com/user-attachments/assets/3fdc95b5-9230-45f1-a5ff-1168b1fb6c25" />

zehjotkah commented:
Thanks, this unfortunately doesn't work for Rybbit. 
There is some kind of problem with the prefixes, so it seems we need to set the backend URL via Coolify's UI and uncheck the "strip prefixes" checkbox. 
But I documented it in the service template docs. It's nearly a one click deploy. The user only has to set both URLs. Maybe @goldflag has some details on how to solve this.

zehjotkah commented:
is there anything I can do to have this merged?

ShadowArcanist commented:
Hey @zehjotkah just now I tested this service and I am not able to create an account.

I can see some 404s on the network tab but no errors on the frontend after clicking "continue" button:
<img width="2559" height="1439" alt="image" src="https://github.com/user-attachments/assets/40089f4d-65e3-48cc-b876-50c3d1a442b7" />

Also the page goes blank randomly and I can see more 404s on network tab
<img width="2560" height="1440" alt="image" src="https://github.com/user-attachments/assets/8df680f8-c16d-4a91-830f-0e75648000b6" />

All the service containers were running healthy and I don't see any error messages on container logs.

Could you please test this template again and check if it works for you?

Also currently the template is using `latest` tag for `image: 'ghcr.io/rybbit-io/rybbit-client:latest'` and `image: 'ghcr.io/rybbit-io/rybbit-backend:latest'` so could you please pin to a static version rather than 'latest' to prevent the template from breaking due to upstream changes.

ShadowArcanist commented on templates/compose/rybbit.yaml:
```diff
@@ -0,0 +1,133 @@
+
+services:
+  rybbit:
+    image: 'ghcr.io/rybbit-io/rybbit-client:latest'
```
Please pin to a static version rather than 'latest' to prevent the template from breaking due to upstream changes.

ShadowArcanist commented on templates/compose/rybbit.yaml:
```diff
@@ -0,0 +1,133 @@
+      start_period: 10s
+
+  rybbit_backend:
+    image: 'ghcr.io/rybbit-io/rybbit-backend:latest'
```
Please pin to a static version rather than 'latest' to prevent the template from breaking due to upstream changes.

zehjotkah commented:
> Hey @zehjotkah just now I tested this service and I am not able to create an account.
> 
> I can see some 404s on the network tab but no errors on the frontend after clicking "continue" button:


@ShadowArcanist just checking to verify: have you set up the domains as per the instructions? (https://github.com/zehjotkah/coolify-docs/blob/next/docs/services/rybbit.md) 

    set frontend URL.
    set backend URL to $frontend_URL/api and uncheck the "Strip Prefixes" option.

zehjotkah committed:
replaced latest tag with specific version

ShadowArcanist commented:
> > Hey @zehjotkah just now I tested this service and I am not able to create an account.
> > I can see some 404s on the network tab but no errors on the frontend after clicking "continue" button:
> 
> @ShadowArcanist just checking to verify: have you set up the domains as per the instructions? (https://github.com/zehjotkah/coolify-docs/blob/next/docs/services/rybbit.md)
> 
> ```
> set frontend URL.
> set backend URL to $frontend_URL/api and uncheck the "Strip Prefixes" option.
> ```

Oopies! I forgot it my bad :/

It's working fine after following those instructions!!

achillion commented:
Please merge already

andrasbacsai commented:
Thank you for the PR!
